File: convalida-validators/src/main/java/convalida/validators/PatternValidator.java
package convalida.validators;

import android.widget.EditText;
import java.util.regex.Pattern;

/**
 * @author Wellington Costa on 21/06/2017.
 */
public class PatternValidator extends AbstractValidator {

    private String pattern;

    public PatternValidator(
            EditText editText,
            String errorMessage,
            String pattern,
            boolean autoDismiss) {
        super(editText, errorMessage, autoDismiss);
        this.pattern = pattern;
    }

    @Override public boolean isValid(String value) {
        return Pattern.matches(pattern, value);
    }

}

File: convalida-validators/src/main/java/convalida/validators/PatternValidator.kt
package convalida.validators

import android.widget.EditText
import java.util.regex.Pattern

/**
 * @author Wellington Costa on 21/06/2017.
 */
class PatternValidator(editText: EditText, errorMessage: String, pattern: String) : AbstractValidator(editText, errorMessage) {

    private var pattern: String? = pattern

    override fun isNotValid(value: String): Boolean {
        return !Pattern.matches(pattern!!, value)
    }

}
